File: arcgis_map_ios/ios/Classes/ZoomStreamHandler.swift
//
//  ZoomStreamHandler.swift
//  arcgis_map_ios
//
//  Created by Julian Bissekkou on 30.11.22.
//

import Foundation

class ZoomStreamHandler: NSObject, FlutterStreamHandler {
    
    private var sink: FlutterEventSink?
    private var lastZoomLevel: Int?
    
    func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        sink = events
        return nil
    }
    
    func onCancel(withArguments arguments: Any?) -> FlutterError? {
        sink = nil
        return nil
    }
    
    func addZoom(zoom: Int) {
        if(lastZoomLevel != zoom) {
            lastZoomLevel = zoom
            sink?(zoom)
        }
    }
}

File: arcgis_map_sdk_android/android/src/main/kotlin/dev/fluttercommunity/arcgis_map_sdk_android/ZoomStreamHandler.kt
package dev.fluttercommunity.arcgis_map_sdk_android

import io.flutter.plugin.common.EventChannel

class ZoomStreamHandler : EventChannel.StreamHandler {
    private var sink: EventChannel.EventSink? = null
    private var lastZoomLevel: Int? = null

    fun addZoom(zoom: Int) {
        if (lastZoomLevel != zoom) {
            lastZoomLevel = zoom
            sink?.success(zoom)
        }
    }

    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
        sink = events
    }

    override fun onCancel(arguments: Any?) {
        sink = null
    }
}